src/wallet/wallet.h

```diff
@@ -1793,11 +1802,13 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     void ReacceptWalletTransactions();
     void ResendWalletTransactions(int64_t nBestBlockTime);
     std::vector<uint256> ResendWalletTransactionsBefore(int64_t nTime);
-    CAmount GetBalance(const isminefilter& filter=ISMINE_SPENDABLE, const int min_depth=0) const;
-    CAmount GetUnconfirmedBalance() const;
-    CAmount GetImmatureBalance() const;
-    CAmount GetUnconfirmedWatchOnlyBalance() const;
-    CAmount GetImmatureWatchOnlyBalance() const;
+    CAmount GetBalance(const std::optional<int>& asOfHeight,
+                       const isminefilter& filter=ISMINE_SPENDABLE,
+                       const int min_depth=0) const;
+    CAmount GetUnconfirmedBalance(const std::optional<int>& asOfHeight) const;
+    CAmount GetImmatureBalance(const std::optional<int>& asOfHeight) const;
+    CAmount GetUnconfirmedWatchOnlyBalance(const std::optional<int>& asOfHeight) const;
```

Comment:
This and `GetUnconfirmedBalance` should both have the `asOfHeight` parameter removed, and should be documented here to make clear that `Unconfirmed` means only transactions in the mempool. Also, do these actually take into account shielded funds belonging to the wallet? If not, then document (and consider renaming to `GetUnconfirmedTransparent*`